src/sql/workbench/browser/modelComponents/table.component.ts

```diff
@@ -116,14 +111,55 @@ export default class TableComponent extends ComponentBase<azdata.TableComponentP
 		}
 	}
 
-	public static transformData(rows: string[][], columns: any[]): { [key: string]: string }[] {
+	private static createIconColumn(col: azdata.TableColumn): Slick.Column<any> {
+		return <Slick.Column<any>>{
+			name: col.name ?? col.value,
+			id: col.value,
+			field: col.value,
+			width: col.width,
+			cssClass: col.cssClass,
+			headerCssClass: col.headerCssClass,
+			toolTip: col.toolTip,
+			formatter: iconCssFormatter,
+			filterable: false
+		};
+	}
+
+	private static createTextColumn(col: azdata.TableColumn): Slick.Column<any> {
+		return <Slick.Column<any>>{
+			name: col.name ?? col.value,
+			id: col.value,
+			field: col.value,
+			width: col.width,
+			cssClass: col.cssClass,
+			headerCssClass: col.headerCssClass,
+			toolTip: col.toolTip,
+			formatter: textFormatter,
+		};
+	}
+
+
+
+	public static transformData(rows: (string | azdata.IconColumnCellValue)[][], columns: any[], iconCssMap?: { [iconKey: string]: string }): { [key: string]: string | { text: string, ariaLabel: string } }[] {
 		if (rows && columns) {
+			let iconCssCache = iconCssMap ?? [];
 			return rows.map(row => {
-				let object: { [key: string]: string } = {};
+				let object: { [key: string]: string | { text: string, ariaLabel: string } } = {};
 				if (row.forEach) {
 					row.forEach((val, index) => {
 						let columnName: string = (columns[index].value) ? columns[index].value : <string>columns[index];
-						object[columnName] = val;
+						if (val['icon']) {
+							const icon: IUserFriendlyIcon = (<azdata.IconColumnCellValue>val).icon;
+							const iconKey: string = getIconKey(icon);
+							const iconCssClass = createIconCssClass(icon, iconCssCache[iconKey]);
+							if (iconCssCache[iconKey] !== iconCssClass) {
+								iconCssCache[iconKey] = iconCssClass;
+							}
+
+							object[columnName] = { text: iconCssClass, ariaLabel: (<azdata.IconColumnCellValue>val).ariaLabel };
```

Comment:
It seems weird to have text be the iconCssClass. There isn't a better value you could use?